### PR TITLE
[libc][NFC] Use user defined literals to build 128 and 256 bit constants.

### PR DESCRIPTION
**Pull request by gchatelet**

### CODE REVIEW

**Comment by gchatelet** [comment]
Most of the change is done via `sed` so it should be safe. I'm running it on an aarch64 machine to make sure all the tests are exercised.

**Comment by lntue** [comment]
I'm sorry for asking a bit late, but is it easy to add `'` to divide 128-bit and 256-bit numbers into 32-bit or 64-bit chunks?

**Comment by gchatelet** [comment]
> I'm sorry for asking a bit late, but is it easy to add `'` to divide 128-bit and 256-bit numbers into 32-bit or 64-bit chunks?

Sure no worries, which one do you want 64 or 32?
Shall I also fix the sage scripts?

**Comment by lntue** [comment]
> > I'm sorry for asking a bit late, but is it easy to add `'` to divide 128-bit and 256-bit numbers into 32-bit or 64-bit chunks?
> 
> Sure no worries, which one do you want 64 or 32? Shall I also fix the sage scripts?

I prefer 32-bit chunks if it does not push the constants to the next line.  Otherwise, 64-bit chunks works fine too.
It would be great for the sage scripts to be updated.  Thanks!

**Review by lntue — APPROVED** [review]
Thanks for cleaning this up!  It looks much better now!

**Comment by gchatelet** [comment]
I used this function and updated the documentation
https://sagecell.sagemath.org/?q=neiiww

I haven't checked the scripts but they should be just a typo away from working 😅

File: libc/src/__support/integer_literals.h [hunk 1]
```diff
@@ -76,7 +76,7 @@ template <typename T, int base> struct DigitBuffer {
   // Returns the digit for a particular character.
   // Returns 255 if the character is invalid.
   LIBC_INLINE static constexpr uint8_t get_digit_value(const char c) {
-    const auto to_lower = [](char c) { return c | 32; };
+    const auto to_lower = [](char c) -> uint8_t { return c | 32; };
     const auto is_digit = [](char c) { return c >= '0' && c <= '9'; };
     const auto is_alpha = [](char c) {
       return ('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z');
```
Comment (nickdesaulniers) [inline libc/src/__support/integer_literals.h:82]: Seems like this is producing an error on the gcc build bots:

```
/home/llvm-libc-buildbot/buildbot-worker/libc-x86_64-debian-fullbuild/libc-x86_64-debian-gcc-fullbuild-dbg/llvm-project/libc/src/__support/integer_literals.h:85:16: error: conversion from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} may change value [-Werror=conversion]
   85 |       return c - '0';
      |              ~~^~~~~
```

Comment (gchatelet) [reply libc/src/__support/integer_literals.h:82]: Yes just noticed, I'll revert, I don't have time to fix it now.

**Pull request merged**